okapi-core/src/test/java/org/folio/okapi/util/ModuleCacheTest.java

```diff
@@ -0,0 +1,350 @@
+  }
+
+  @Test
+  void testLookupEmpty() {
+    Map<String, List<ModuleCache.ModuleCacheEntry>> map = new HashMap<>();
+
+    assertThat(ModuleCache.lookup("", HttpMethod.GET, map, true, null)).isEmpty();
+    assertThat(ModuleCache.lookup("/", HttpMethod.GET, map, true, null)).isEmpty();
+    assertThat(ModuleCache.lookup("/a", HttpMethod.GET, map, true, null)).isEmpty();
+    assertThat(ModuleCache.lookup("/a/b", HttpMethod.GET, map, true, null)).isEmpty();
+  }
+
+  @Test
+  void testLookupRoutingEntries() {
+    ModuleDescriptor md = new ModuleDescriptor();
+    md.setId("module-1.0.0");
+    List<RoutingEntry> routingEntries = new LinkedList<>();
+    RoutingEntry routingEntry1 = new RoutingEntry();
+    routingEntry1.setPathPattern("/a/b");
+    routingEntry1.setMethods(new String[] {"GET"});
+    routingEntries.add(routingEntry1);
+    RoutingEntry routingEntry2 = new RoutingEntry();
+    routingEntry2.setPathPattern("/a/b");
+    routingEntry2.setMethods(new String[] {"POST"});
+    routingEntries.add(routingEntry2);
+    RoutingEntry routingEntry3 = new RoutingEntry();
+    routingEntry3.setPathPattern("/a/b/{id}/c");
+    routingEntry3.setMethods(new String[] {"GET"});
+    routingEntries.add(routingEntry3);
+    RoutingEntry routingEntry4 = new RoutingEntry();
+    routingEntry4.setPathPattern("/p/*/y");
+    routingEntry4.setMethods(new String[] {"GET"});
+    routingEntries.add(routingEntry4);
+    RoutingEntry routingEntry5 = new RoutingEntry();
+    routingEntry5.setPathPattern("/perms/users*");
+    routingEntry5.setMethods(new String[] {"GET"});
+    routingEntries.add(routingEntry5);
+    RoutingEntry routingEntry6 = new RoutingEntry();
+    routingEntry6.setPath("/old/type");
+    routingEntry6.setMethods(new String[] {"GET"});
+    routingEntries.add(routingEntry6);
+
+    Map<String, List<ModuleCache.ModuleCacheEntry>> map = new HashMap<>();
+    ModuleCache.add(md, map, routingEntries);
+
+    assertThat(ModuleCache.lookup("", HttpMethod.GET, map, true, null)).isEmpty();
+    assertThat(ModuleCache.lookup("/", HttpMethod.GET, map, true, null)).isEmpty();
+    assertThat(ModuleCache.lookup("/a", HttpMethod.GET, map, true, null)).isEmpty();
+    List<ModuleInstance> instances = ModuleCache.lookup("/a/b", HttpMethod.GET, map, true, null);
+    assertThat(instances.size()).isEqualTo(1);
+    assertThat(instances.get(0).getRoutingEntry()).isEqualTo(routingEntry1);
+
+    instances = ModuleCache.lookup("/a/b", HttpMethod.GET, map, true, "module-1.0.0");
+    assertThat(instances.size()).isEqualTo(1);
+    assertThat(instances.get(0).getRoutingEntry()).isEqualTo(routingEntry1);
+
+    assertThat(ModuleCache.lookup("/a", HttpMethod.GET, map, true, "other-1.0.0")).isEmpty();
+
+    instances = ModuleCache.lookup("/a/b", HttpMethod.POST, map, true, null);
+    assertThat(instances.size()).isEqualTo(1);
+    assertThat(instances.get(0).getRoutingEntry()).isEqualTo(routingEntry2);
```

Comment:
If you have

    Function<ModuleInstance, RoutingEntry> routingEntry = ModuleInstance::getRoutingEntry;

you can reduce the two lines to a one-liner:

    assertThat(instances).extracting(routingEntry).containsExactly(routingEntry1);

Reply:
ok.. I can tell you.. You'll be able to find many constructs  - for example in ModuleTest - that could be simplified.

Reply:
Thanks, more readable now.